compiler/rustc_errors/src/lib.rs

```diff
@@ -187,29 +187,40 @@ impl SubstitutionPart {
 impl CodeSuggestion {
     /// Returns the assembled code suggestions, whether they should be shown with an underline
     /// and whether the substitution only differs in capitalization.
-    pub fn splice_lines(&self, sm: &SourceMap) -> Vec<(String, Vec<SubstitutionPart>, bool)> {
+    pub fn splice_lines(
+        &self,
+        sm: &SourceMap,
+    ) -> Vec<(String, Vec<SubstitutionPart>, Vec<Vec<(usize, usize)>>, bool)> {
```

Comment:
It's not clear what `Vec<Vec<(usize, usize)>>` means, probably better to make a struct with named fields for `(usize, usize)`.

Reply:
`Vec< Vec<(usize, usize) >>`:

 - First `Vec` corresponds to each line of the resulting suggestion
 - Second `Vec` corresponds to each highlighted replacement of a multipart suggestion
 - the first `usize` is the start of the pseudo-span to be highlighted
 - the second `usize` is the end of the pseudo-span

Should I also somehow make the `Vec<Vec<_>>` hierarchy clearer? I hoped that the _uses_ would be enough, but maybe embedded comments, at least, would help too?

Reply:
A single comment on `fn splice_lines` explaining what it returns should be enough.